### PR TITLE
Update claim account check

### PR DESCRIPTION
**Pull request by asalzmann**
## Context and purpose of the change

`BaseAccount` and `StridePeriodicVestingAccount` types should be able to claim airdrops.

## Brief Changelog

- Check for `BaseAccount` and `StridePeriodicVestingAccount` types explicitly
- Add tests

## Author's Checklist

I have...

- [ ] Run and PASSED locally all GAIA integration tests
- [ ] If the change is contentful, I either:
    - [x] Added a new unit test OR 
    - [ ] Added test cases to existing unit tests
- [ ] OR this change is a trivial rework / code cleanup without any test coverage

If skipped any of the tests above, explain.

## Reviewers Checklist

*All items are required. Please add a note if the item is not applicable and please add
your handle next to the items reviewed if you only reviewed selected items.*

I have...

- [ ] reviewed state machine logic
- [ ] reviewed API design and naming
- [ ] manually tested (if applicable)
- [ ] confirmed the author wrote unit tests for new logic
- [ ] reviewed documentation exists and is accurate

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes? 
  - [ ] Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`?
  - [ ] This pull request updates existing proto field values (and require a backend and frontend migration)? 
  - [ ] Does this pull request change existing proto field names (and require a frontend migration)?
  How is the feature or change documented? 
      - [ ] not applicable
      - [ ] jira ticket `XXX` 
      - [ ] specification (`x/<module>/spec/`) 
      - [ ] README.md 
      - [ ] not documented

### CODE REVIEW

File: x/claim/keeper/claim.go [hunk 1]
```diff
@@ -152,7 +150,7 @@ func (k Keeper) GetAirdropByIdentifier(ctx sdk.Context, airdropIdentifier string
 func (k Keeper) GetDistributorAccountBalance(ctx sdk.Context, airdropIdentifier string) (sdk.Coin, error) {
 	airdrop := k.GetAirdropByIdentifier(ctx, airdropIdentifier)
 	if airdrop == nil {
-		return sdk.Coin{}, errorsmod.Wrapf(sdkerrors.ErrInvalidAddress, "invalid airdrop identifier: GetDistributorAccountBalance")
+		return sdk.Coin{}, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid airdrop identifier: GetDistributorAccountBalance")
```
Comment (shellvish) [inline x/claim/keeper/claim.go:153]: Should we use `errorsmod` or `sdkerrors` in general? I see [other files](https://github.com/Stride-Labs/stride/blob/main/x/stakeibc/keeper/icacallbacks_reinvest.go) use `errorsmod`, so just want to make sure this is intentional!

File: x/claim/keeper/claim.go [hunk 2]
```diff
@@ -505,11 +515,13 @@ func (k Keeper) ClaimCoinsForAction(ctx sdk.Context, addr sdk.AccAddress, action
 		return nil, err
 	}
 
-	// If the account is a default vesting account, don't grant new vestings
+	// Only BaseAccounts and StridePeriodicVestingAccount can claim
 	acc := k.accountKeeper.GetAccount(ctx, addr)
-	_, isDefaultVestingAccount := acc.(*authvestingtypes.BaseVestingAccount)
-	if isDefaultVestingAccount {
-		return nil, err
+	_, isStrideVestingAccount := acc.(*vestingtypes.StridePeriodicVestingAccount)
```
Comment (shellvish) [inline x/claim/keeper/claim.go:520]: I didn't realize you can do this to check types, very neat

**Review by shellvish — APPROVED** [review]
Good catch, this fix looks great! Quick nit on error types and we're set.

File: x/claim/keeper/claim.go (hunk 1 above)
Comment (sampocs) [reply x/claim/keeper/claim.go:153]: We should use errorsmod anytime we’re calling Wrapf or Register

sdkerrors should be used when referring to an error type (such as sdkerrors.ErrInvalidAddress)

Comment (asalzmann) [reply x/claim/keeper/claim.go:153]: Updated - what's the rationale for using one vs. the other out of curiosity?

Comment (sampocs) [reply x/claim/keeper/claim.go]: sdkerrors.Wrapf and sdkerrors.Register are deprecated! So they’ll cause lint errors which will block the PR from being merged l.

similar to sdk.Int → sdkmath.Int

**Pull request merged**